client/blocks/plugins-update-manager/schedule-form.helper.ts

```diff
@@ -1,104 +1,99 @@
-export const MAX_SELECTABLE_PLUGINS = 10;
+/**
+ * Prepare unix timestamp in seconds
+ * based on selected frequency, day, hour and period
+ */
+export const prepareTimestamp = (
+	frequency: string,
+	day: string,
+	hour: string,
+	period: string
+) => {
+	const event = new Date();
 
-export const DAILY_OPTION = {
-	label: 'Daily',
-	value: 'daily',
+	const hours = parseInt( hour ) + ( period === 'pm' ? 12 : 0 );
+	event.setHours( hours, 0, 0, 0 );
+
+	if ( frequency === 'daily' ) {
+		// Set next day
+		event.setDate( event.getDate() + 1 );
+	}
+
+	if ( frequency === 'weekly' ) {
+		// Set first next selected day
+		event.setDate( event.getDate() + ( ( parseInt( day ) + 7 - event.getDay() ) % 7 || 7 ) );
+	}
+
+	// return timestamp in seconds
+	return event.getTime() / 1000;
+};
+
+/**
+ * Validate name
+ * - required
+ * - max length 120
+ */
+export const validateName = ( name: string ) => {
+	let error = '';
+	if ( ! name ) {
+		error = 'Please provide a name to this plugin update schedule.';
+	} else if ( name.length > 120 ) {
+		error = 'Please provide a shorter name.';
+	}
+
+	return error;
 };
 
-export const WEEKLY_OPTION = {
-	label: 'Weekly',
-	value: 'weekly',
+type TimeSlot = {
+	frequency: string;
+	timestamp: number;
 };
+/**
+ * Validate time slot
+ * based on existing schedules in context of frequency
+ */
+export const validateTimeSlot = ( newSchedule: TimeSlot, existingSchedules: TimeSlot[] = [] ) => {
+	let error = '';
+	const newDate = new Date( newSchedule.timestamp * 1000 );
 
-export const DAY_OPTIONS = [
-	{
-		label: 'Monday',
-		value: 'a',
-	},
-	{
-		label: 'Tuesday',
-		value: 'b',
-	},
-	{
-		label: 'Wednesday',
-		value: 'c',
-	},
-	{
-		label: 'Thursday',
-		value: 'd',
-	},
-	{
-		label: 'Friday',
-		value: 'e',
-	},
-	{
-		label: 'Saturday',
-		value: 'f',
-	},
-	{
-		label: 'Sunday',
-		value: 'g',
-	},
-];
+	existingSchedules.forEach( ( schedule ) => {
+		if ( error ) {
+			return;
+		}
 
-export const HOUR_OPTIONS = [
-	{
-		label: '00',
-		value: '0',
-	},
-	{
-		label: '01',
-		value: '1',
-	},
-	{
-		label: '02',
-		value: '3',
-	},
-	{
-		label: '03',
-		value: '3',
-	},
-	{
-		label: '04',
-		value: '4',
-	},
-	{
-		label: '05',
-		value: '5',
-	},
-	{
-		label: '06',
-		value: '6',
-	},
-	{
-		label: '07',
-		value: '7',
-	},
-	{
-		label: '08',
-		value: '8',
-	},
-	{
-		label: '09',
-		value: '9',
-	},
-	{
-		label: '10',
-		value: '10',
-	},
-	{
-		label: '11',
-		value: '11',
-	},
-];
+		const existingDate = new Date( schedule.timestamp * 1000 );
 
-export const PERIOD_OPTIONS = [
-	{
-		label: 'AM',
-		value: 'am',
-	},
-	{
-		label: 'PM',
-		value: 'pm',
-	},
-];
+		if (
+			( newSchedule.frequency === 'daily' || schedule.frequency === 'daily' ) &&
+			existingDate.getHours() === newDate.getHours()
+		) {
+			error = 'Please pick another time for optimal performance, as this slot is already taken.';
```

Comment:
```suggestion
			error = 'Please choose another time, as this slot is already scheduled.';
```
Maybe something like this?

Reply:
I agree; it sounds better.
I have followed Figma and didn't think too much about the copy since all copies will be reviewed soon.